conjure-java-core/src/test/java/com/palantir/conjure/java/types/WireFormatTests.java

```diff
@@ -561,6 +567,52 @@ public void testOptionalAliasExample() throws IOException {
                 .isEqualTo(OptionalAliasExample.of(OptionalAlias.of(Optional.empty())));
     }
 
+    @Test
+    public void testEmptyOptionalFieldSerialization() throws IOException {
+        assertThat(mapper.writeValueAsString(PrimitiveOptionalsExample.builder()
+                        .aliasTwo(StringAliasTwo.of(Optional.empty()))
+                        .aliasOptional(OptionalAlias.of(Optional.empty()))
+                        .aliasOptionalMap(OptionalMapAliasExample.of(Optional.empty()))
+                        .build()))
+                .isEqualTo("{}");
+    }
+
+    @Test
+    public void testEmptyOptionalFieldSerialization_nonEmptyValues() throws IOException {
+        PrimitiveOptionalsExample original = PrimitiveOptionalsExample.builder()
+                .num(OptionalDouble.of(0D))
+                .bool(Optional.of(Boolean.FALSE))
+                .integer(OptionalInt.of(0))
+                .map(ImmutableMap.of())
+                .list(ImmutableList.of())
+                .set(ImmutableSet.of())
+                .aliasOne(StringAliasOne.of(""))
+                .aliasTwo(StringAliasTwo.of(Optional.of(StringAliasOne.of(""))))
+                .aliasOptional(OptionalAlias.of(Optional.of("")))
+                .aliasOptionalMap(OptionalMapAliasExample.of(Optional.of(ImmutableMap.of())))
+                .aliasMap(MapAliasExample.of(ImmutableMap.of()))
+                .build();
+        PrimitiveOptionalsExample recreated =
+                mapper.readValue(mapper.writeValueAsBytes(original), PrimitiveOptionalsExample.class);
+        assertThat(recreated).isEqualTo(original);
```

Comment:
For documentation purposes really, can you also add a test that shows the actual string serialized output, like in the above test?

Reply:
That sort of test can be difficult to maintain because we don't provide any guarantees about field order, which doesn't impact tests using zero or one fields.
How about adding the serialized json string to the failure message if the test doesn't pass, and verifying that the expected number of fields are present?

Reply:
Alternatively a test per field, that way you don't have any ordering issues, but that sounds like a lot of faff. Should we be specifying field order, I think jackson supports that right?

Reply:
We could provide consistent order, but I don't think the conjure spec should. Not sure its worth it just for this sort of test, as written I think it should give us sufficient context.

Reply:
ok let's ship it. I still feel like consistent order might be useful for these sort of tests, whenever I see serialization code, I want to see the ACTUAL string/bytes. Roundtrip tests are obviously useful, not denying that, but nothing beats just seeing the actual output.